sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2935,6 +2935,16 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)
 
+  val ADD_DEFAULTS_FOR_INSERTS_WITHOUT_USER_SPECIFIED_FIELDS =
+    buildConf("spark.sql.defaultColumn.addDefaultsForInsertsWithoutUserSpecifiedFields")
+      .internal()
+      .doc("When true, for each INSERT command without any user-specified fields where the " +
```

Comment:
What does `user-specified fields` mean here?

Reply:
So the configuration is to fill in default values. If `spark.sql.defaultColumn.useNullsForMissingDefaultValues` and one of the filled column doesn't have a default value, there will be error, right?

Reply:
Good questions:

* The `user-specified fields` are explicit columns after the table in the FROM clause, e.g. when the command looks like `INSERT INTO mytable(columnA, columnB) VALUES (...))`.
* If `spark.sql.defaultColumn.useNullsForMissingDefaultValues` is true and one of these missing columns does not have an explicit DEFAULT value, then NULL will be appended. Otherwise, the command will fail with an error message reporting that the number of columns provided is incorrect.

I added this information to the config description here.